bears/vcs/git/GitCommitBear.py

```diff
@@ -24,4 +25,36 @@ def get_remotes():
 
     def get_head_commit(self):
         with change_directory(self.get_config_dir() or os.getcwd()):
+            command = self.check_github_pull_request_temporary_merge_commit()
+            if command:
+                return run_shell_command(command)
+
             return run_shell_command('git log -1 --pretty=%B')
+
+    def check_github_pull_request_temporary_merge_commit(self):
+        """
+        This detects the merge commit generated by GitHub
```

Comment:
Also, I think this function doesn't just detect, but also creates a git command, so a sentence like below
`This function creates a git command to fetch the unmerged parent commit shortlog from a commit generated by GitHub in a refs/pull....`
might be better to explain it..